src/layers/conv.jl

```diff
@@ -36,9 +35,22 @@ GCNConv(ch::Pair{Int,Int}, σ = identity; kwargs...) =
 
 @functor GCNConv
 
+# function (l::GCNConv)(fg::FeaturedGraph, x::AbstractMatrix)
+#     L̃ = normalized_laplacian(fg, eltype(x); selfloop=true)
+#     l.σ.(l.weight * x * L̃ .+ l.bias)
+# end
+
+message(l::GCNConv, xi, xj) = xj
```

Comment:
I am quite confused with `GCNConv` implemented with message-passing neural network. I doesn't leverage anything from the MPNN. As I said, all it need are algebraic operations, e.g. `l.weight * x .+ l.bias`, instead of indexing by `xi` or `xj`. So, I don't think `GCNConv <: MessagePassing` is good. Furthermore, people wouldn't classify `GCNConv` as a spatial-based graph convolution layer. It is a spectral-based graph convolution layer.

Reply:
> I am quite confused with GCNConv implemented with message-passing neural network. I doesn't leverage anything from the MPNN.

It is leveraging `gather` and `scatter(+,...)`, which basically implement sparse matrix multiplication

> As I said, all it need are algebraic operations, e.g. l.weight * x .+ l.bias, instead of indexing by xi or xj. So, I don't think GCNConv <: MessagePassing is good. Furthermore, people wouldn't classify GCNConv as a spatial-based graph convolution layer. It is a spectral-based graph convolution layer.

Although it was introduced as a spectral-based operator, I think most people these days think about it in the general message passing scheme. In any case, both interpretations are valid. The important thing is to do something which is well adapted to the underlying graph implementation. Calling `adjacency_matrix(fg)` is a huge performance hit when the underlying is COO or adjacency list, we have to prohibit these perfomance traps. 
I plan to expand FeaturedGraph support for sparse adjacency matrix representations,  so that we can have back the algebraic form of GCNConv.  I would do that in a separate PR though if you don't mind, this is already quite huge

Reply:
> I think most people these days think about it in the general message passing scheme.

Sorry, I couldn't accept this statement. It is definition. Please don't break definitions. If most people do, they are wrong in the beginning.

> The important thing is to do something which is well adapted to the underlying graph implementation.

I agree with this statement.

> Calling `adjacency_matrix(fg)` is a huge performance hit when the underlying is COO or adjacency list, we have to prohibit these perfomance traps.

Under definition, it is no way to take the performance issue as excuse. If the huge performance is such, then just find a way to resolve the conversion penalty issues, rather than breaking definitions.

Reply:
> I plan to expand FeaturedGraph support for sparse adjacency matrix representations

I agree with you on this point and this is the way I planned long time ago.

> I would do that in a separate PR though if you don't mind, this is already quite huge

Yeah, I have the same feeling. It's huge. Let's break it down.